zapcore/hook_test.go

```diff
@@ -71,3 +71,47 @@ func TestHooks(t *testing.T) {
 		}
 	}
 }
+
+func TestHooksWithFields(t *testing.T) {
+	tests := []struct {
+		entryLevel Level
+		coreLevel  Level
+		expectCall bool
+	}{
+		{DebugLevel, InfoLevel, false},
+		{InfoLevel, InfoLevel, true},
+		{WarnLevel, InfoLevel, true},
+	}
+
+	for _, tt := range tests {
+		fac, logs := observer.New(tt.coreLevel)
+		intField := makeInt64Field("foo", 42)
+		intField2 := makeInt64Field("bar", 67)
+		ent := Entry{Message: "bar", Level: tt.entryLevel}
+
+		var called int
+		f := func(e Entry, fs []Field) error {
```

Comment:
we're not verifying fields, we should verify that the fields match the logger that's used (ideally with different loggers that have different fields attached)